app/vmselect/main.go

```diff
@@ -213,6 +213,18 @@ func RequestHandler(w http.ResponseWriter, r *http.Request) bool {
 	}
 
 	if strings.HasPrefix(path, "/vmalert") {
+		if strings.HasSuffix(path, "/") {
```

Comment:
This will result in two redirects when `/vmalert/` path is requested:

1. `/vmalert/` -> `/vmalert`
2. `/vmalert` -> `/vmalert/home`

The code also theoretically may improperly work for urls starting with `/vmalert` and ending with `/vmalert`. For example:
- `/vmalertmanager/foo` would be proxied to vmalert, while it shouldn't be proxied there.
- `/vmalert/foo/vmalert` would be improperly redirected to `/vmalert/foo/vmalert/home`

Note also that it is better to use `http.StatusMovedPermanently` instead of `http.StatusFound` for this type of redirect, so user agents (e.g. web browsers) could automatically add missing `/` to the end of the url without the need to send the request to `vmselect`. See https://stackoverflow.com/questions/1393280/http-redirect-301-permanent-vs-302-temporary 
It is better to use the following code instead:

```go
if path == "/vmalert" {
  http.Redirect(w, r, path+"/", http.StatusMovedPermanently)
}
if strings.HasPrefix(path, "/vmalert/") {
   // The rest of the code starting from the line 228
}
```

Pay attention to the